src/realm/sync/client.cpp

```diff
@@ -1600,8 +1656,10 @@ void SessionWrapper::actualize(ServerEndpoint endpoint)
         }
     }
 
-    if (!m_client_reset_config)
-        report_progress(); // Throws
+    if (!m_client_reset_config && m_reliable_download_progress) {
```

Comment:
So looking at the origins of this line. It got added in this PR https://github.com/realm/realm-sync/pull/2779, and I think it was to resolve this issue https://github.com/realm/realm-sync/issues/2859. I think the idea is that we want to report that the session has been activated and has received zero progress so that apps can start a spinner/progress bar in their UI. Why do we only want to call report_progress() here if we have reliable download progress?

Reply:
I'm not sure that is the case here since essentially to understand the behaviour of this one would need to look at both sides: sync client impl and object store notifier. Looks like it's changed a few time (or fixed) but it's hard to tell what was the intent really. I think https://github.com/realm/realm-sync/issues/2859 is different issue: it essentially asks to report progress the moment client is about to start downloading the changeset every time. I don't think it makes sense to do it on this level in client.cpp, but anyway the issue is still open. The suggested change fixing this was probably irrelevant.
Toggle to m_reliable_download_progress == true happens only once on connect in current implementation, so my guess is that it was there to report initial sync status on connect with changesets to upload. I don't think it is useful in any other way.
I believe I've changed it now to be explicit about how it behaved before. I don't see anything breaking with this or changing observed behaviour. What else should be done here? What do you think?